pyccel/ast/internals.py

```diff
@@ -163,3 +162,46 @@ def __str__(self):
             stop = str(self.stop)
         return '{0} : {1}'.format(start, stop)
 
+class Symbol(Basic):
+    """
+        #TODO
+    """
+    is_Symbol = True
+
+    def __init__(self, name):
+        if not isinstance(name, str):
+            raise TypeError('Symbol name should be a string, not '+ str(type(name)))
+        self._name = name
+        super().__init__()
+
+    @property
+    def name(self):
+        return self._name
+
+    def __eq__(self, other):
+        if self is other:
+            return True
+        if type(self) is type(other):
+            try :
+                return self._name == other.name
+            except:
+                return False
```

Comment:
Didn't you want to be able to compare with strings?
Can't you just do:
`return str(self)==s`
?

Reply:
Or:
`return self._name == s`

Reply:
I was just afraid that some were in the code we use __eq__ between two objects of different class

Reply:
If we do then they can always write their own `__eq__` method which handles strings. If it's a different class then the definition of `==` might be more complicated than just having the same name anyway

Reply:
I'm lost can you explain more what do you expect, if I only compare the property name it can be another object with the same name and has the property, and then it will be equal.
that's why I check the type too, this is also helping with the class that inherits from Symbol like (Variable)